### PR TITLE
feat(GlobalHeader): Added the better type for the link handler

### PR DESCRIPTION
**Pull request by MattSchiller**
### Overview

Added more specific typing to the AppHeader.props.onLinkAction item param

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut-labs/src/GlobalHeader/types.tsx [hunk 1]
```diff
@@ -13,7 +16,7 @@ type BaseHeader = {
   /** A method to be called on click/activating a header item */
   action: AppHeaderClickHandler;
   /** A method to be called only on click/activating a *link* header item */
-  onLinkAction?: AppHeaderClickHandler;
+  onLinkAction?: AppHeaderClickHandler<AppHeaderItemWithHref>;
```
Comment (MattSchiller) [inline packages/gamut-labs/src/GlobalHeader/types.tsx:19]: I'd like to do `item.href` in my `onLinkAction` method without having to cast it in the parent, so adding it here.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.52-alpha.5a4318.0<br/>@codecademy/gamut-labs@19.12.2-alpha.5a4318.0<br/>@codecademy/styleguide@41.10.22-alpha.5a4318.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60d6430543e895202f96aa62--gamut-preview.netlify.app>https://60d6430543e895202f96aa62--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60d6430543e895202f96aa62>Deploy Logs</a></p>

**Review by JoshuaKGoldberg — APPROVED** [review]

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.53-alpha.ea9ed7.0<br/>@codecademy/gamut-labs@19.12.3-alpha.ea9ed7.0<br/>@codecademy/styleguide@41.10.23-alpha.ea9ed7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60d9e2bd2f18ae0abb4b3577--gamut-preview.netlify.app>https://60d9e2bd2f18ae0abb4b3577--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60d9e2bd2f18ae0abb4b3577>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.53-alpha.68d75e.0<br/>@codecademy/gamut-labs@19.12.3-alpha.68d75e.0<br/>@codecademy/styleguide@41.10.23-alpha.68d75e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60d9e53a09557bed5deca934--gamut-preview.netlify.app>https://60d9e53a09557bed5deca934--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60d9e53a09557bed5deca934>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.4.54-alpha.1b95e3.0<br/>@codecademy/gamut-labs@19.12.4-alpha.1b95e3.0<br/>@codecademy/styleguide@41.10.24-alpha.1b95e3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60d9e7b4d8eda91ef5b4d2d4--gamut-preview.netlify.app>https://60d9e7b4d8eda91ef5b4d2d4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60d9e7b4d8eda91ef5b4d2d4>Deploy Logs</a></p>

**Pull request merged**